src/main/java/datawave/iterators/FrequencyMetadataAggregator.java

```diff
@@ -0,0 +1,414 @@
+package datawave.iterators;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import org.apache.accumulo.core.data.ByteSequence;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.IteratorEnvironment;
+import org.apache.accumulo.core.iterators.LongCombiner;
+import org.apache.accumulo.core.iterators.OptionDescriber;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.iterators.WrappingIterator;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.WritableUtils;
+import org.apache.log4j.Logger;
+
+import datawave.marking.MarkingFunctions;
+import datawave.query.model.DateFrequencyMap;
+
+/**
+ * Aggregates entries in the metadata table for the "f", "i", and "ri" columns. When initially ingested, entries for these columns have a column qualifier with
+ * the format {@code <datatype>\0<yyyyMMdd>}, and a value containing a possibly partial frequency count for the date in the column qualifier. Entries with the
+ * same row, column family, datatype, and column family will be aggregated into a single entry where the column qualifier consists of the datatype and the value
+ * consists of an encoded {@link DateFrequencyMap} with the dates and counts seen. Additionally, this aggregator will handle the case where we have a previously
+ * aggregated entry and freshly ingested rows that need to be aggregated together. <br>
+ * <br>
+ * This iterator supports the following options:
+ * <ul>
+ * <li>{@value COMBINE_VISIBILITIES}: Defaults to false. If true, entries will be aggregated by row, column family, and datatype only, and the column visibility
+ * will be a combination of all column visibilities seen for the row/column family/datatype combo. This option is meant to be used when scanning only, and not
+ * for compaction.</li>
+ * </ul>
+ */
+public class FrequencyMetadataAggregator extends WrappingIterator implements OptionDescriber {
+    
+    public static final String COMBINE_VISIBILITIES = "FrequencyMetadataAggregator.COMBINE_VISIBILITIES";
```

Comment:
Lets just make this COMBINE_VISIBILITIES.  Since we are setting this for this class, the owner of the option will be obvious

Reply:
Done.